### PR TITLE
chore: Add pre-release issue and PR templates

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds two different issue/PR templates:

- The issue template can be used in _this_ repository to track the Rust toolchain updates
- The PR template can be used in downstream product operators to track the Rust dependency bumps. This PR template can be linked in the tracking issue located in the "issues" repository.

# Issue Template Preview

## Pre-Release Rust Toolchain Update for Stackable Release XX.(X)X

Part of <https://github.com/stackabletech/issues/TRACKING_ISSUE>

> [!NOTE]
> During a Stackable release we need to ensure that every product operator uses the latest Rust toolchain (used by us). To keep the toolchain in sync across all our operators, we update the version centrally in this repository.

### Update Rust Toolchain

```[tasklist]
### Tasks in this Repository
- [ ] Update Rust toolchain in the `config/rust.yaml` file.
- [ ] Generate downstream PRs using the ["Generate Downstream PRs"](https://github.com/stackabletech/operator-templating/actions/workflows/generate_prs.yml) action.
- [ ] Merge downstream PRs, see below for more details.
```

### Merge Downstream PRs

Replace the items in the task lists below with the applicable Pull Requests

```[tasklist]
### Tasks in Downstream Repositories
- [ ] https://github.com/stackabletech/airflow-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/commons-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/druid-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/hbase-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/edc-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/hdfs-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/hello-world-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/hive-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/kafka-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/nifi-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/listener-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/opa-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/secret-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/spark-k8s-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/superset-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/trino-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
- [ ] https://github.com/stackabletech/zookeeper-operator/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files
```

---

# PR Template Preview

## Bump Rust Dependencies for Stackable Release XX.(X)X

Part of <https://github.com/stackabletech/issues/TRACKING_ISSUE>

> [!NOTE]
> During a Stackable release we need to update various Rust dependencies before entering the final release period to ensure we run the latest versions of crates. These bumps also include previously updated and released crates from the `operator-rs` repository.

```[tasklist]
### Tasks
- [ ] Bump Rust Dependencies, see below for more details.
- [ ] Add changelog entry stating which important crates were bumped (including the version).
```

> [!NOTE]
> The bumping / updating of Rust dependencies is done in multiple steps:
>
> 1. Update the minimum Version in the root `Cargo.toml` manifest.
> 2. Run the `cargo update` command, which also updates the `Cargo.lock` file.
> 3. Lastly, run `make regenerate-nix` to update the `Cargo.nix` file.

```[tasklist]
### Bump Rust Dependencies
- [ ] Bump `stackable-operator` and friends.
- [ ] Bump `product-version`.
- [ ] Bump all other dependencies.
```

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pre-release.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### Merge Downstream PRs
+
+Replace the items in the task lists below with the applicable Pull Requests
+
+<!--
+    The following list was generated by:
+
+    yq '.repositories[].name' config/repositories.yaml \
+    | xargs -I {} echo "- [ ] https://github.com/stackabletech/{}/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files"
```
Comment (NickLarsenNZ) [inline .github/ISSUE_TEMPLATE/pre-release.md:44]: I've been adding `sort` to mine just to ensure consistent ordering (eg: the yaml parsing will be in order of entry and not alphabetical).

```suggestion
    yq '.repositories[].name' config/repositories.yaml \
    | sort \
    | xargs -I {} echo "- [ ] https://github.com/stackabletech/{}/pulls?q=sort:updated-desc+is:pr+is:open+Update+templated+files"
```

File: template/.github/PULL_REQUEST_TEMPLATE/pre-release.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+> During a Stackable release we need to update various Rust dependencies before
+> entering the final release period to ensure we run the latest versions of
+> crates. These bumps also include previously updated and released crates from
+> the `operator-rs` repository.
+
+```[tasklist]
+### Tasks
+- [ ] Bump `stackable-operator` and friends.
+- [ ] Bump `product-version`.
+- [ ] Bump all other dependencies.
```
Comment (NickLarsenNZ) [inline template/.github/PULL_REQUEST_TEMPLATE/pre-release.md:19]: Shall we mention how? And including the nix deps?

```suggestion
- [ ] Bump all other dependencies with `cargo update` and `make regenerate-nix`.
```

Or leave it to a help section at the bottom (which could also mention bumping minimum versions in `cargo.toml`?

**Review by NickLarsenNZ — APPROVED** [review]
Wanted to also mention if we should add a note about `pre-commit`, but couldn't find an appropriate place, so maybe that's a separate thing.

File: .github/ISSUE_TEMPLATE/pre-release.md (hunk 1 above)
Comment (Techassi) [reply .github/ISSUE_TEMPLATE/pre-release.md:44]: Good point!

File: template/.github/PULL_REQUEST_TEMPLATE/pre-release.md (hunk 2 above)
Comment (Techassi) [reply template/.github/PULL_REQUEST_TEMPLATE/pre-release.md:19]: I think a more thorough how-to / help section might be better here.

Comment (Techassi) [reply template/.github/PULL_REQUEST_TEMPLATE/pre-release.md:32]: Mentioned in b4974c5

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Comment by Techassi** [comment]
Personal reminder: This still needs to be rolled-out in downstream repos.

**Pull request merged**